src/util/mongo.controller.js

```diff
@@ -131,6 +133,33 @@ const updateBulkExportStatus = async (clientId, newStatus, error = null) => {
   }
 };
 
+/**
+ * Changes the first valid request time for tracking 429:TooManyRequests errors
+ * @param {string} clientId the id of the client making the export request
+ * @param {Object} timeOfFirstValidRequest a Date object storing the time of the first valid request
+ */
+const updateFirstValidRequest = async (clientId, timeOfFirstValidRequest) => {
```

Comment:
These multiple update calls are a bit of over engineering in my humble opinion. They could be single update function with arguments for the two timeout tracking fields. But this is just a nitpick and this is just our engineering test server so I wouldn't undo the work here. It makes the other logic look cleaner anyways.